lib/protobuf/enum.rb

```diff
@@ -120,7 +127,8 @@ def self.enum_for_name(name)
     #   Enums, the first enum defined will be returned.
     #
     def self.enum_for_tag(tag)
-      enums_for_tag(tag).first
+      value = mapped_enums[tag.to_i]
```

Comment:
Nice catch, if you look at the diff for `self.enums_for_tag`, it was calling `tag.to_i` already. Looks like this might be a pre-existing bug?